app/script/main/app.js

```diff
@@ -756,14 +756,15 @@ z.main.App = class App {
         }
 
         const baseUrl = z.util.Environment.frontend.isLocalhost() ? '/page/auth.html' : '/auth/';
-        let url = `${baseUrl}${location.search}`;
+        let url = `${baseUrl}${window.location.search}`;
         const isImmediateSignOutReason = App.CONFIG.SIGN_OUT_REASONS.IMMEDIATE.includes(signOutReason);
         if (isImmediateSignOutReason) {
           url = z.util.URLUtil.appendParameter(url, `${z.auth.URLParameter.REASON}=${signOutReason}`);
         }
 
         const redirectToLogin = signOutReason !== z.auth.SIGN_OUT_REASON.NOT_SIGNED_IN;
-        if (redirectToLogin) {
+        const forcedAppLogin = z.util.Environment.desktop && z.window.location.hash === '#login';
```

Comment:
What is that `z.window` I don't find it in the codebase?

Reply:
Isn't it supposed to be a `||` here?
If the app is the desktop or that the current location contains `login`?

Reply:
Just a typo, already fixed

Reply:
I want to keep the '#login' only if it is in the context of the app.